type-check/join.test.ts

```diff
@@ -0,0 +1,18 @@
+import * as Test from "../src/types/Test";
+import { toAsync, join, pipe } from "../src";
+
+const { checks, check } = Test;
+
+const res1 = join("-", []);
+const res2 = join("!", [1, 2, 3, 4]);
+const res3 = join("*", toAsync([1, 2, 3, 4]));
+const res4 = pipe([1, 2, 3, 4], join("-"));
+const res5 = pipe(toAsync([1, 2, 3, 4]), join("#"));
+
+checks([
+  check<typeof res1, [], Test.Pass>(),
```

Comment:
`join` only returns string. Type of `res1` must be empty string, not an empty array.
```suggestion
heck<typeof res1, "", Test.Pass>(),
```

Reply:
I fixed it :>